sql/src/main/java/io/crate/expression/scalar/geo/CoordinateFunction.java

```diff
@@ -31,13 +30,20 @@
 import java.util.List;
 import java.util.function.Function;
 
+import static io.crate.metadata.functions.Signature.scalar;
+import static io.crate.types.TypeSignature.parseTypeSignature;
+
 public final class CoordinateFunction {
 
-    private static final List<DataType> SUPPORTED_INPUT_TYPES = ImmutableList.of(DataTypes.GEO_POINT, DataTypes.STRING, DataTypes.DOUBLE_ARRAY, DataTypes.UNDEFINED);
+    private static final List<DataType<?>> SUPPORTED_INPUT_TYPES =
+        List.of(DataTypes.GEO_POINT, DataTypes.STRING, DataTypes.DOUBLE_ARRAY, DataTypes.UNDEFINED);
 
     private static void register(ScalarFunctionModule module, String name, Function<Object, Double> func) {
-        for (DataType inputType : SUPPORTED_INPUT_TYPES) {
-            module.register(new UnaryScalar<>(name, inputType, DataTypes.DOUBLE, func));
+        for (DataType<?> inputType : SUPPORTED_INPUT_TYPES) {
+            module.register(
+                scalar(name, inputType.getTypeSignature(), parseTypeSignature("double precision")),
```

Comment:
what is the preferable way of defining the signature
```
parseTypeSignature("double precision")
```
or 
```
DataTypes.DOUBLE.getTypeSignature()
```
?

Reply:
No strict preference, both is valid. I prefer `parseTypeSignature("double precision")` as I think is more readable, but may be just a personal preference.

Reply:
Wouldn't the second variant be safer, since the type name must be valid to create the `DataType` instance later on?

Reply:
`parseTypeSignature()` will also fail on invalid types as it will call `DataTypes.ofName()` eventually.

Reply:
> parseTypeSignature() will also fail on invalid types as it will call DataTypes.ofName() eventually.

Good point, although I think `DataTypes.DOUBLE.getTypeSignature()`  or `parseTypeSignature(DataTypes.DOUBLE.getName())`  would make the relationship clearer and make refactorings easier.

Reply:
Refactoring is a good point although I really like the literal variant. But yes lets use then `DataTypes.DOUBLE.getTypeSignature()` in future.